packages/SwingSet/docs/configuration.md

```diff
@@ -133,17 +184,78 @@ not yet supported.
 
 - `enablePipelining` specifies whether the vat should be launched with
-  promise pipelining enabled (defaults to `false` if omitted).
+  promise pipelining enabled.  Defaults to `false`.
+
+- `description` is an optional description string displayed in log messages for
+  debugging purposes.  If omitted, the kernel will synthesize a description
+  string from some of the other options data.
+
+- `name` is an optional short label string used to identify the vat in log and
+  debug outputs.  If omitted, it defaults to the property name that was used for
+  the vat descriptor.
+
+- `enableDisavow`, if `true`, adds `vatPowers.disavow()`, which allows vat code
+  to explicitly disavow interest in an imported Presence. This will trigger a
+  `syscall.dropImports` of the associated object ID. By default, this
+  function is not added to `vatPowers`.
+
+- `managerType` indicates what sort of vat worker to run the associated vat in.
+  If omitted, it defaults to the swingset's default manager type.
 
 - `metered` specifies whether the vat should be have metering turned
-  on or not (defaults to `false` if omitted).
+  on or not. Defaults to `false`.
+
+- `reapInterval` specifies the reap interval for this particular vat.  It
+  defaults to the swingset's default reap interval.
 
 - `enableSetup` indicates that the vat module may use the older, lower
   level `setup()` API, which allows a vat to be defined independent of
   the liveslots framework (defaults to `false` if omitted).  (If you
   don't understand what this means, do not use this.)
 
-- `enableDisavow`, if true, adds `vatPowers.disavow()`, which allows vat code
-  to explicitly disavow interest in an imported Presence. This will trigger a
-  `syscall.dropImports` of the associated object ID. By default, this
-  function is not added to `vatPowers`.
+- `critical`, if `true`, marks the vat as critical to the correct functioning of
+  the swingset.  If a critical vat suffers a failure, the kernel will panic and
+  the swingset as a whole will be terminated without saving any intermediate
+  state that resulted during the crank that caused the vat to die.  Defaults to
+  `false`.
+
+- `useTranscript`, if `true`, says the vat should record a transcript of all
+  deliveries made to it so that these can be replayed at a later time to
+  reconstruct the internal state of the vat.  Defaults to `true`.
+
+- `virtualObjectCacheSize` tells the vat's virtual object manager how many
+  virtual objects should have their state kept live in memory at any given
+  time.  Defaults to an implementation defined value.
+
+## Dynamic option setting
+
+Some swingset and vat configuration options can be dynamically updated within a
+running swingset.
+
+### Dynamic kernel options
+
+Swingset-wide (kernel) options are updated by calling the
+`changeKernelOptions(options)` method on the swingset's controller object, where
+`options` is an object indicating which options to change and the values to
+change them to.
+
```

Comment:
Add:

`controller.changeKernelOptions(options)` should be called when the kernel is idle, i.e. not in the middle of a `controller.run()`. Modifying options will cause state changes to be written to the block buffer (i.e. the `hostStorage` object provided to `buildVatController`), and the host application is solely responsible for committing those changes to persistent storage, so most applications will need to do a `commit()` some time after calling `controller.changeKernelOptions()`.